### PR TITLE
RabbitTemplate - Fix resolution of mq connection factory… (#657)

### PR DESCRIPTION
**Pull request by celper**
`RabbitTemplate.ConvertSendAndReceive()` threw exception when used with Simple RoutingConnectionFactory.

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/SteeltoeOSS/Steeltoe?pullRequest=691) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/SteeltoeOSS/Steeltoe?pullRequest=691) before we can accept your contribution.<br/><br/>:x: celper [sign now](https://cla.dotnetfoundation.org/SteeltoeOSS/Steeltoe?pullRequest=691)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/SteeltoeOSS/Steeltoe?pullRequest=691) it.</sub>

File: src/Messaging/test/RabbitMQ.Test/Core/RabbitTemplateSimpleRoutingConnectionFactoryTest.cs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+using Steeltoe.Messaging.RabbitMQ.Connection;
+using System;
+using Xunit;
+using RC = RabbitMQ.Client;
+
+namespace Steeltoe.Messaging.RabbitMQ.Core
+{
+    [Trait("Category", "Integration")]
+    public class RabbitTemplateSimpleRoutingConnectionFactoryTest
+        : IDisposable
```
Comment (kenswan) [inline src/Messaging/test/RabbitMQ.Test/Core/RabbitTemplateSimpleRoutingConnectionFactoryTest.cs:15]:
```suggestion
```

File: src/Messaging/test/RabbitMQ.Test/Core/RabbitTemplateSimpleRoutingConnectionFactoryTest.cs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            channel1.Verify(c => c.BasicPublish("exchFoo", "rkFoo", It.IsAny<bool>(), It.IsAny<RC.IBasicProperties>(), It.IsAny<byte[]>()));
+
+            cf2.Verify(cf => cf.CreateConnection(), Times.AtLeastOnce);
+            channel2.Verify(c => c.BasicPublish("exchBar", "rkBar", It.IsAny<bool>(), It.IsAny<RC.IBasicProperties>(), It.IsAny<byte[]>()));
+        }
+
+
+        public void Dispose()
+        {
+        }
```
Comment (kenswan) [inline src/Messaging/test/RabbitMQ.Test/Core/RabbitTemplateSimpleRoutingConnectionFactoryTest.cs:85]:
```suggestion
```

**Comment by kenswan** [comment]
@celper looks like there's just a couple extra blank lines to remove (reported by the azure pipeline check) along with the unused `dispose`. Outside of that, looks great!

File: src/Messaging/test/RabbitMQ.Test/Core/RabbitTemplateSimpleRoutingConnectionFactoryTest.cs (hunk 2 above)
Comment (celper) [reply src/Messaging/test/RabbitMQ.Test/Core/RabbitTemplateSimpleRoutingConnectionFactoryTest.cs]: fixed

**Comment by kenswan** [comment]
Thanks for the contribution @celper!

**Pull request merged**